### PR TITLE
Side modal ref tabel can collapse

### PR DESCRIPTION
**Pull request by chinook25**

### CODE REVIEW

File: apps/molgenis-components/src/components/tables/RefTable.vue [hunk 1]
```diff
@@ -1,66 +1,119 @@
...
+        </button>
       </span>
+    </h5>
+    <div>
+      <div :class="{ 'collapsed-table': collapsed }">
+        <table class="table table-sm mb-2">
+          <tr v-for="(value, key) in filteredResults(reference)">
+            <td class="key border-right">{{ key }}</td>
+            <td class="value">
+              <div v-if="reference.metadata">
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/RefTable.vue:24]: metadata is mandatory so this if can be removed

File: apps/molgenis-components/src/components/tables/RefTable.vue [hunk 2]
```diff
@@ -1,66 +1,119 @@
...
 }>();
 
-function filteredResults(reference: IRefModalData): IRefModalData {
-  const filtered = { ...reference };
-  delete filtered.name;
+const canCollapse = Object.keys(filteredResults(reference)).length > 3;
+let collapsed = ref(startsCollapsed);
+if (collapsed && !canCollapse) {
+  collapsed.value = false;
+}
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/RefTable.vue:102]: this should be part of the initialisation e.g. ref(startCollapsed && canCollapse);

**Review by mswertz — CHANGES_REQUESTED** [review]
Looks great. 

For 'ref' and 'ontology' I propose we don't collapse.

For 'ref_array' and 'ontology_array' I propose that you collapse when [more than one item] AND [number of columns > 5] 

I also found a bug, see 
image

To get this view go to 
https://preview-emx2-pr-2179.dev.molgenis.org/catalogue/tables/#/CollectionEvents

File: apps/molgenis-components/src/components/tables/DataDisplayCell.vue [hunk 3]
```diff
@@ -52,10 +52,10 @@ export default defineComponent({
     cellTypeComponentName() {
       return this.isArrayType
         ? "ListDisplay"
-        : typeMap[this.metaData.columnType] || "StringDisplay";
+        : typeMap[this.metaData?.columnType] || "StringDisplay";
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/DataDisplayCell.vue]: metadata is required, so I do not approve of this monkey patch. This should be fixed at the input side.

File: apps/molgenis-components/src/components/tables/RefTable.vue [hunk 4]
```diff
@@ -94,6 +94,9 @@ let filteredResults = computed(() => getFilteredResults(reference.value));
 let canCollapse = computed(() => Object.keys(filteredResults.value).length > 3);
 
 let collapsed = ref(startsCollapsed.value && canCollapse.value);
+let primaryKey = computed(() =>
+  getPrimaryKey(reference.value, reference.value.metadata)
+);
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/RefTable.vue:99]: minor thingy, let's put computeds together if possible, in order to keep to files more clean.

File: apps/molgenis-components/src/components/tables/TableExplorer.vue [hunk 5]
```diff
@@ -338,14 +338,13 @@
...
       </p>
     </ConfirmModal>
-
     <RefSideModal
       v-if="refSideModalProps"
       :table-id="refSideModalProps?.table"
       :label="refSideModalProps?.label"
       :rows="refSideModalProps?.rows"
+      :schema="refSideModalProps?.schema"
       @onClose="refSideModalProps = undefined"
```
Comment (chinook25) [inline apps/molgenis-components/src/components/tables/TableExplorer.vue]: questionmarks here are not necessary (because of the v-if)

**Review by mswertz — APPROVED** [review]

**Pull request merged**